compiles and runs successfully

maryamariyan opened the pull request.

maryamariyan committed:
Adding a portion of System.ComponentModel.Composition tests

danmoseley commented on src/System.ComponentModel.Composition/src/Strings.resx:
```diff
@@ -0,0 +1,153 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
We should not check in any strings.designer.cs files

Generally there should be exactly one strings.resx file and it should be in src/Resources, not here. Please check...

danmoseley replied:
Also neither strings.resx nor strings.designer.cs should be in the project file at all. strings.resx is automatically found in the resources folder

danmoseley commented on src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/CompositionContractMismatchException.cs:
```diff
@@ -56,6 +57,16 @@ public CompositionContractMismatchException(string message, Exception innerExcep
+        {
+        }
+
 #if FEATURE_SERIALIZATION
```
Remove FEATURE_SERIALIZATION symbol everywhere.
Retain the content if it is in an exception type or tests for one. Delete it if it is not in an exception type.

danmoseley commented on src/System.ComponentModel.Composition/src/Microsoft/Internal/Assumes.cs:
```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Delete this and use C:\git\corefx\src\Common\src\Microsoft\Internal\Assumes.cs

danmoseley replied:
and other assumes file

danmoseley commented on src/System.ComponentModel.Composition/tests/System/ComponentModel/IO/FileIO.cs:
```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
these files in /IO/*cs duplicate the existing test helpers for creating and using temp files/directories.
leave as is for now but in future someone could eliminate the duplication

danmoseley replied:
You have checked in 2 copies of them all though. You only need 1 copy. Please check for dupe C# files

danmoseley commented on src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/Hosting/TypeCatalogTests.cs:
```diff
@@ -0,0 +1,574 @@
+            return assembly;
+        }
+
+#if FEATURE_INTERNAL_REFLECTIONCONTEXT
```
We don't define any defines, generally. So for each one you must remove it, and make a decision whether to keep the code inside it. If they existed in the implementation, do the same thing we did for that. There shold probably be zero `#if` in the code.

danmoseley commented on src/System.ComponentModel.Composition/tests/IExportableTest.cs:
```diff
@@ -0,0 +1,12 @@
+using System;
```
Many of your files are missing the license. To check, do something liek this:
`for /f %i in ('dir /s/b *') do @(@echo %i & head -4 %i)`

I got `head` etc from the GNU tools for Win32 or you can use bash.

Always look for ways to get a simple script to do the hard work for you :)

maryamariyan replied:
I'll make sure to address this as well. thanks

maryamariyan committed:
removing FEATURE_SERIALIZATION blocks for exception classes keeping the code it wraps

maryamariyan committed:
Removing FEATURE_SERIALIZATION blocks for exception classes and removing the code it wraps

danmoseley commented:
You can make a PR..

maryamariyan committed:
Exposing #if code blocks used in this project, in this commit.
Todo: need to decide which of the following constants are assumed to be true: FEATURE_TRACING, MEF_FEATURE_REGISTRATION, FEATURE_ADVANCEDCOLLECTIONS, FEATURE_FILEIO.

maryamariyan committed:
fixing build

maryamariyan committed:
Switching Fact skip to Fact with active issue

maryamariyan committed:
Deleting files that havent yet been added to project

maryamariyan committed:
Adding license header to missing files

maryamariyan committed:
Deleting files that havent yet been added to project

maryamariyan committed:
Remove and sorting using statements

maryamariyan committed:
Removing MEF_FEATURE_REGISTRATION condition. Not keeping the code it wraps

maryamariyan committed:
Removing AssumesTests. Will add in the next PR.

maryamariyan committed:
To enable the remaining code blocks wrapped by FEATURE_SERIALIZATION, we need fix on corefx for 20519

maryamariyan committed:
Removing extra blank lines

maryamariyan committed:
Using Assert.All rather than foreach

maryamariyan committed:
Setting proper work item issue number

maryamariyan committed:
Enable FEATURE_ADVANCEDCOLLECTIONS

maryamariyan committed:
Keeping FEATURE_TRACING as it is also available in System.Composition.Convention.

Enabled in System.ComponentModel.Composition but disabled in System.Composition.Convention

maryamariyan committed:
Enabling all code paths wrapped in FEATURE_SERIALIZATION condition

maryamariyan committed:
Merge branch 'master' into port-mef-tests

ViktorHofer committed:
Revert "Enabling all code paths wrapped in FEATURE_SERIALIZATION condition"

This reverts commit 42230c427f3bacc61c7881bd0b5ce2adf687a014.

maryamariyan committed:
Will not make ComposablePartException and ComposableException serializable

maryamariyan committed:
Adding missing TypeForward and platform support for netfx

ViktorHofer committed:
Add tests for S.C.Composition. Netfx blob aren't correct yet

ViktorHofer committed:
Merge branch 'port-mef-tests' of https://github.com/maryamariyan/corefx into port-mef-tests

maryamariyan committed:
Revert "Revert "Enabling all code paths wrapped in FEATURE_SERIALIZATION condition""

This reverts commit b936192b83d309fc0fbbdb93673f6f87b683c88c.

ViktorHofer committed:
Setting configurations properly and adding correct netfx blobs for serialization testing

maryamariyan committed:
Fixing MissingManifestResourceException by removing RootNamespace

maryamariyan committed:
Removing SerializableCompositionElement, keeping its test class only for netfx

maryamariyan committed:
Removing SerializableCompositionElementTest for netfx too.

ViktorHofer committed:
Fix serialization remoteexecutor bug, remove vs autogenerated comments, fix wrong blobs

ViktorHofer committed:
Remove unnecessary facade for ref

maryamariyan committed:
Fixing indentation

ViktorHofer committed:
Fix RemoteExecutor reference

ViktorHofer committed:
Merge branch 'port-mef-tests' of https://github.com/maryamariyan/corefx into port-mef-tests
